app/scripts/services/metrics.js

```diff
@@ -169,6 +169,46 @@ angular.module("openshiftConsole")
       });
     };
 
+    // Returns custom metrics available for a particular pod
+    var getCustomMetrics = function(pod) {
+      var namespace = pod.metadata.namespace;
+      var podName = pod.metadata.name;
+
+      return getMetricsURL().then(function(metricsURL) {
+
+        if (!metricsURL) {
+          return null;
+        }
+
+        var url = metricsURL + "/metrics";
+
+        var params = {
+          tags: "custom_metric:true,pod_name:" + podName
+        };
+
+        return $http.get(url, {
+          auth: {},
+          headers: {
+            Accept: 'application/json',
+            'Hawkular-Tenant': namespace
+          },
+          params: params
+        }).then(function(response) {
+          var metrics = [];
+          angular.forEach(response.data, function(value) {
+            var metric = {
+              name: value.tags.metric_name,
+              unit: value.tags.units,
+              description: value.tags.description,
+            };
+            metrics.push(metric);
+          });
```

Comment:
Suggest using `_.map`. (Untested, but I think this will work.)

```js
return _.map(response.data, function(value) {
  return {
    name: value.tags.metric_name,
    unit: value.tags.units,
    description: value.tags.description
  };
});
```